### ISSUE TITLE
[PyTorch] weight tensor assumption

### ISSUE THREAD

**Issue by SsnL** [pytorch/pytorch#7865]
Now in both inits and spectral norm, we assume that dim `0` of `weight` is the output feature dim. However, this is incorrect for `ConvTranspose`. 

We need to figure out how to properly handle those.

**Comment by t-vi** [comment]
I'll look into this together with #8208.
Quite likely we supply the dimension of the output with a default of 0?

**Comment by SsnL** [comment]
@t-vi Yes you are correct. Feel free to change the init methods to also take in a `dim` arg, but that can be done after SN.

**Comment by SsnL** [comment]
Thank you!

**Comment by t-vi** [comment]
Should this be closed or is there something we yet want to implement?
